texastribune/tx_salaries
# Cleanup Dallas department names

[GenericDepartmentMixin](https://github.com/texastribune/tx_salaries/blob/master/tx_salaries/utils/transformers/mixins.py#L63-L70) introduced some grammar errors to City of Dallas departments. We generally want to present exactly what the agency gave, but in this case the transformer added errors that didn’t exist before. (see [screenshot](https://cloud.githubusercontent.com/assets/1087467/7544027/3f79f93e-f57e-11e4-95d1-49acd437f84a.jpg))
#### What's the plan?

Remove department name mixin from the Dallas transformer, use the department name cleaver and add some regular expressions to correct capitalization and spelling:
- fix “Police Uinform” spelling
- convert “&” to “and” (handled by name [cleaver](https://github.com/texastribune/tx_salaries/blob/master/tx_salaries/utils/cleaver.py#L18-L19))
- convert “’S” to “’s” (handled by name cleaver)
- convert “Mcc” to “MCC”
- convert “Pom” to “POM”
#### How should this be manually tested?

`manage.py import_salary_data /path/to/dallas-salary-spreadsheet.xlsx`
check the department names (see [screenshot](https://cloud.githubusercontent.com/assets/1087467/7544035/5e29ff1e-f57e-11e4-9ddd-29048e4ff53d.png))
#### Next steps?

Mixins probably need better documentation for writing transformers
